File: 50pow(x, n)/pow(x,n).cpp
class Solution {
public:
    double myPow(double x, int n) {
        auto result{1.};
        auto const sign{std::signbit(n)};
        while (n)
        {
            if (n & 1) result *= x;
            x *= x;
            n /= 2;
        }
        return sign ? 1 / result : result;
    }
};

File: 50pow(x, n)/pow(x,n).java
class Solution {
    public double myPow(double x, int n) {
        double result = 1;
        final double sign = Math.signum(n);
        while (n != 0)
        {
            if ((n & 1) != 0) result *= x;
            x *= x;
            n /= 2;
        }
        return sign == 1 ? result : 1 / result;
    }
}
